pkg/gktest/assertion.go

```diff
@@ -92,20 +100,28 @@ func (a *Assertion) matchesCountStr(matching int32) error {
 	switch a.Violations.StrVal {
 	case "yes":
 		if matching == 0 {
+			if a.Message != nil {
+				return fmt.Errorf("%w: got %d violations containing %q but want at least %d",
+					ErrNumViolations, matching, *a.Message, 1)
+			}
 			return fmt.Errorf("%w: got %d violations but want at least %d",
 				ErrNumViolations, matching, 1)
 		}
 
 		return nil
 	case "no":
 		if matching > 0 {
+			if a.Message != nil {
+				return fmt.Errorf("%w: got %d violations containing %q but want none",
+					ErrNumViolations, matching, *a.Message)
+			}
 			return fmt.Errorf("%w: got %d violations but want none",
 				ErrNumViolations, matching)
 		}
 
 		return nil
 	default:
-		return fmt.Errorf(`%w: assertion.violation, if set, must be an integer, "yes", or "no"`,
+		return fmt.Errorf(`%w: assertion.violation, if set, must be a nonnegative integer, "yes", or "no"`,
```

Comment:
This seem error text is used twice.  Should it be a custom error type like `ErrInvalidYAML`?

Reply:
I don't think so - the unit tests cover this path and it's valid for someone to refactor how this logic is implemented so this error is returned in a single location. If this error text was needed in more places (e.g. when we increase the complexity of assertions) then it would make sense. But I'd rather wait until then.